lib/streamlit/DeltaGenerator.py

```diff
@@ -1876,6 +1881,39 @@ def slider(
             current_value = current_value[0] if single_value else current_value
         return current_value if single_value else tuple(current_value)
 
+    @_with_element
+    def file_uploader(self, element, label, type=['*'], key=None):
+
+        """Display a file uploader widget.
+
+        Parameters
+        ----------
+        label : str or None
+            A short label explaining to the user what this file uploader is for.
+        type : string[] or None
+            Array of allowed extensions. (['.png', '.jpg'])
+            Default value: ['*']
+
+        Returns
+        -------
+        byte[] or None
+            The byte array of uploaded file or None if no one file is loaded.
+
+        Examples
+        --------
+        >>> file = st.file_uploader("Upload a image", type=([".png"]))
```

Comment:
Should be `type=[".png"]`

But actually, can we change `type` so:
* It works if you pass a `str`. (That is, it should accept both `str` and `iterable`)
* You don't need to pass in the `.` in `.png`